lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -244,16 +243,17 @@ public static Domain getDomain(
     private static Domain getDomain(
             ColumnDescriptor column,
             Type type,
-            List<Object> minimums,
-            List<Object> maximums,
+            ValuesList valuesList,
             boolean hasNullValue,
             DateTimeZone timeZone)
     {
-        checkArgument(minimums.size() == maximums.size(), "Expected minimums and maximums to have the same size");
-
         if (type.equals(BOOLEAN)) {
-            boolean hasTrueValues = minimums.stream().anyMatch(value -> (boolean) value) || maximums.stream().anyMatch(value -> (boolean) value);
-            boolean hasFalseValues = minimums.stream().anyMatch(value -> !(boolean) value) || maximums.stream().anyMatch(value -> !(boolean) value);
+            if (valuesList instanceof DiscreteValuesList) {
+                throw new VerifyException("Encountered dictionary encoding for boolean");
```

Comment:
why not?

Reply:
Parquet encoding does not support dictionary for booleans (see `ParquetEncoding#initDictionary`)